### PR TITLE
Milvus, Chroma, Qdrant, Pinecone: Update CDK

### PR DESCRIPTION
**Pull request by flash1293**
## What

Updates CDK for Milvus, Chroma, Qdrant and Pinecone connectors to get fix in for Azure embedders.

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  destination-qdrant test report (commit [`118dfae4e2`](https://github.com/airbytehq/airbyte/pull/30820/commits/118dfae4e20f02610112b7231ea0c8ce9b745a60)) - ✅

⏲️  Total pipeline duration: 03mn46s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build destination-qdrant docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for destination-qdrant                        | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1695899064/118dfae4e20f02610112b7231ea0c8ce9b745a60/destination-qdrant/0.0.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:118dfae4e20f02610112b7231ea0c8ce9b745a60)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-qdrant test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-chroma test report (commit [`118dfae4e2`](https://github.com/airbytehq/airbyte/pull/30820/commits/118dfae4e20f02610112b7231ea0c8ce9b745a60)) - ❌

⏲️  Total pipeline duration: 04mn04s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build destination-chroma docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Code format checks                                              | ❌        |
| Validate metadata for destination-chroma                        | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1695899064/118dfae4e20f02610112b7231ea0c8ce9b745a60/destination-chroma/0.0.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:118dfae4e20f02610112b7231ea0c8ce9b745a60)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-chroma test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-pinecone test report (commit [`118dfae4e2`](https://github.com/airbytehq/airbyte/pull/30820/commits/118dfae4e20f02610112b7231ea0c8ce9b745a60)) - ✅

⏲️  Total pipeline duration: 04mn06s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Connector package install                                         | ✅        |
| Build destination-pinecone docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                        | ✅        |
| Integration tests                                                 | ✅        |
| Acceptance tests                                                  | ✅        |
| Code format checks                                                | ✅        |
| Validate metadata for destination-pinecone                        | ✅        |
| Connector version semver check                                    | ✅        |
| Connector version increment check                                 | ✅        |
| QA checks                                                         | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1695899064/118dfae4e20f02610112b7231ea0c8ce9b745a60/destination-pinecone/0.0.14/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:118dfae4e20f02610112b7231ea0c8ce9b745a60)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-pinecone test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-milvus test report (commit [`118dfae4e2`](https://github.com/airbytehq/airbyte/pull/30820/commits/118dfae4e20f02610112b7231ea0c8ce9b745a60)) - ✅

⏲️  Total pipeline duration: 04mn08s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build destination-milvus docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Integration tests                                               | ✅        |
| Acceptance tests                                                | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for destination-milvus                        | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1695899064/118dfae4e20f02610112b7231ea0c8ce9b745a60/destination-milvus/0.0.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:118dfae4e20f02610112b7231ea0c8ce9b745a60)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-milvus test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-qdrant test report (commit [`40af6af121`](https://github.com/airbytehq/airbyte/pull/30820/commits/40af6af121fcce6fc3332e010e78233023c090db)) - ✅

⏲️  Total pipeline duration: 01mn45s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build destination-qdrant docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for destination-qdrant                        | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1695899899/40af6af121fcce6fc3332e010e78233023c090db/destination-qdrant/0.0.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:40af6af121fcce6fc3332e010e78233023c090db)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-qdrant test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-milvus test report (commit [`40af6af121`](https://github.com/airbytehq/airbyte/pull/30820/commits/40af6af121fcce6fc3332e010e78233023c090db)) - ✅

⏲️  Total pipeline duration: 02mn54s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build destination-milvus docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Integration tests                                               | ✅        |
| Acceptance tests                                                | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for destination-milvus                        | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1695899899/40af6af121fcce6fc3332e010e78233023c090db/destination-milvus/0.0.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:40af6af121fcce6fc3332e010e78233023c090db)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-milvus test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-pinecone test report (commit [`40af6af121`](https://github.com/airbytehq/airbyte/pull/30820/commits/40af6af121fcce6fc3332e010e78233023c090db)) - ✅

⏲️  Total pipeline duration: 02mn54s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Connector package install                                         | ✅        |
| Build destination-pinecone docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                        | ✅        |
| Integration tests                                                 | ✅        |
| Acceptance tests                                                  | ✅        |
| Code format checks                                                | ✅        |
| Validate metadata for destination-pinecone                        | ✅        |
| Connector version semver check                                    | ✅        |
| Connector version increment check                                 | ✅        |
| QA checks                                                         | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1695899899/40af6af121fcce6fc3332e010e78233023c090db/destination-pinecone/0.0.14/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:40af6af121fcce6fc3332e010e78233023c090db)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-pinecone test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-chroma test report (commit [`40af6af121`](https://github.com/airbytehq/airbyte/pull/30820/commits/40af6af121fcce6fc3332e010e78233023c090db)) - ✅

⏲️  Total pipeline duration: 03mn09s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build destination-chroma docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for destination-chroma                        | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1695899899/40af6af121fcce6fc3332e010e78233023c090db/destination-chroma/0.0.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:40af6af121fcce6fc3332e010e78233023c090db)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-chroma test
```

File: airbyte-integrations/connectors/destination-chroma/destination_chroma/destination.py [hunk 1]
```diff
@@ -39,7 +47,10 @@ class DestinationChroma(Destination):
     embedder: Embedder
 
     def _init_indexer(self, config: ConfigModel):
-        self.embedder = embedder_map[config.embedding.mode](config.embedding)
+        if config.embedding.mode == "azure_openai" or config.embedding.mode == "openai":
+            self.embedder = embedder_map[config.embedding.mode](config.embedding, config.processing.chunk_size)
+        else:
+            self.embedder = embedder_map[config.embedding.mode](config.embedding)
```
Comment (pedroslopez) [inline airbyte-integrations/connectors/destination-chroma/destination_chroma/destination.py:53]: could you share more about what's going on here and why only chroma needs this?

Comment (pedroslopez) [reply airbyte-integrations/connectors/destination-chroma/destination_chroma/destination.py:53]: oh - are we just now adding the embedder for chroma but others have it already?

File: airbyte-integrations/connectors/destination-chroma/setup.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 from setuptools import find_packages, setup
 
 MAIN_REQUIREMENTS = [
-    "airbyte-cdk[vector-db-based]",
+    "airbyte-cdk[vector-db-based]==0.51.22",
```
Comment (pedroslopez) [inline airbyte-integrations/connectors/destination-chroma/setup.py:9]: Do we need to pin exact? If we want to at least include this version but go up to 0.51.x we could use `~=`

**Review by pedroslopez — APPROVED** [review]
nits/Qs, but LGTM!

File: airbyte-integrations/connectors/destination-chroma/destination_chroma/destination.py (hunk 1 above)
Comment (flash1293) [reply airbyte-integrations/connectors/destination-chroma/destination_chroma/destination.py:53]: yeah, exactly, this is aligning with the others.

File: airbyte-integrations/connectors/destination-chroma/setup.py (hunk 2 above)
Comment (flash1293) [reply airbyte-integrations/connectors/destination-chroma/setup.py:9]: As the vector db CDK currently changes dramatically even in patch versions, I would like to keep it like this for now until things are more stable-

**Comment by airbyte-oss-build-runner** [comment]
##  destination-pinecone test report (commit [`1c2c5848b6`](https://github.com/airbytehq/airbyte/pull/30820/commits/1c2c5848b69e5c73881213820affd87608e732f0)) - ✅

⏲️  Total pipeline duration: 03mn46s 

| Step                                                              | Result   |
|:------------------------------------------------------------------|:---------|
| Connector package install                                         | ✅        |
| Build destination-pinecone docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                        | ✅        |
| Integration tests                                                 | ✅        |
| Acceptance tests                                                  | ✅        |
| Code format checks                                                | ✅        |
| Validate metadata for destination-pinecone                        | ✅        |
| Connector version semver check                                    | ✅        |
| Connector version increment check                                 | ✅        |
| QA checks                                                         | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1696169521/1c2c5848b69e5c73881213820affd87608e732f0/destination-pinecone/0.0.14/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1c2c5848b69e5c73881213820affd87608e732f0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-pinecone test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-qdrant test report (commit [`1c2c5848b6`](https://github.com/airbytehq/airbyte/pull/30820/commits/1c2c5848b69e5c73881213820affd87608e732f0)) - ✅

⏲️  Total pipeline duration: 04mn43s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build destination-qdrant docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for destination-qdrant                        | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1696169521/1c2c5848b69e5c73881213820affd87608e732f0/destination-qdrant/0.0.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1c2c5848b69e5c73881213820affd87608e732f0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-qdrant test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-chroma test report (commit [`1c2c5848b6`](https://github.com/airbytehq/airbyte/pull/30820/commits/1c2c5848b69e5c73881213820affd87608e732f0)) - ✅

⏲️  Total pipeline duration: 04mn44s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build destination-chroma docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for destination-chroma                        | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1696169521/1c2c5848b69e5c73881213820affd87608e732f0/destination-chroma/0.0.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1c2c5848b69e5c73881213820affd87608e732f0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-chroma test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-milvus test report (commit [`1c2c5848b6`](https://github.com/airbytehq/airbyte/pull/30820/commits/1c2c5848b69e5c73881213820affd87608e732f0)) - ✅

⏲️  Total pipeline duration: 05mn06s 

| Step                                                            | Result   |
|:----------------------------------------------------------------|:---------|
| Connector package install                                       | ✅        |
| Build destination-milvus docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                      | ✅        |
| Integration tests                                               | ✅        |
| Acceptance tests                                                | ✅        |
| Code format checks                                              | ✅        |
| Validate metadata for destination-milvus                        | ✅        |
| Connector version semver check                                  | ✅        |
| Connector version increment check                               | ✅        |
| QA checks                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/flash1293_update-connectors/1696169521/1c2c5848b69e5c73881213820affd87608e732f0/destination-milvus/0.0.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1c2c5848b69e5c73881213820affd87608e732f0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-milvus test
```

**Pull request merged**